ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -734,6 +717,11 @@ void open(CompactionInfo ci) throws TException {
         throw new TException("Unable to acquire lock(S) on " + ci.getFullPartitionName());
       }
       lockId = res.getLockid();
+
+      long txnTimeout = MetastoreConf.getTimeVar(conf, MetastoreConf.ConfVars.TXN_TIMEOUT, TimeUnit.MILLISECONDS);
+      Thread beater = new CompactionHeartbeater(this, TxnUtils.getFullTableName(ci.dbname, ci.tableName), conf);
```

Comment:
Nit: I usually try to use the temp local variables immediately after creating.

Reply:
Sure